plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java

```diff
@@ -57,12 +63,21 @@
 import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class HiveModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     @Override
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
-        configBinder(binder).bindConfig(HiveConfig.class);
+        HiveConfig hiveConfig = buildConfigObject(HiveConfig.class);
+        hiveConfig.getGlueProxyApiId().ifPresent(glueProxyApiId -> {
+            GlueHiveMetastoreConfig glueConfig = buildConfigObject(GlueHiveMetastoreConfig.class);
+            checkState(
+                    glueConfig.getGlueRegion().isPresent() && glueConfig.getGlueEndpointUrl().isPresent(),
+                    "Both Glue region and Glue endpoint URL must be provided when Glue proxy API ID is present");
```

Comment:
We can use `@AssertTrue` or `@PostConstruct` annotations within the config class likes `MySqlJdbcConfig` & `BigQueryConfig` for verifying the config values once you moved to Glue config class.

Reply:
Fixed.